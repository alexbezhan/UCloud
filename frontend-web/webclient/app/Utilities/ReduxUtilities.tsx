import activity from "Activity/Redux/ActivityReducer";
import * as AppRedux from "Applications/Redux";
import analyses from "Applications/Redux/AnalysesReducer";
import applications from "Applications/Redux/BrowseReducer";
import detailedApplicationSearch from "Applications/Redux/DetailedApplicationSearchReducer";
import * as TaskRedux from "BackgroundTasks/redux";
import dashboard from "Dashboard/Redux/DashboardReducer";
import {initObject} from "DefaultObjects";
import detailedFileSearch from "Files/Redux/DetailedFileSearchReducer";
import fileInfo from "Files/Redux/FileInfoReducer";
import header, {CONTEXT_SWITCH, USER_LOGIN, USER_LOGOUT} from "Navigation/Redux/HeaderReducer";
import sidebar from "Navigation/Redux/SidebarReducer";
import status from "Navigation/Redux/StatusReducer";
import notifications from "Notifications/Redux/NotificationsReducer";
import * as ProjectRedux from "Project/Redux";
import {Action, AnyAction, combineReducers, createStore, Store} from "redux";
import {composeWithDevTools} from "redux-devtools-extension";
import {createResponsiveStateReducer, responsiveStoreEnhancer} from "redux-responsive";
import simpleSearch from "Search/Redux/SearchReducer";
import {responsiveBP} from "ui-components/theme";
import uploader from "Uploader/Redux/UploaderReducer";
import avatar from "UserSettings/Redux/AvataaarReducer";
import hookStore from "Utilities/ReduxHooks";

export function configureStore(
    initialObject: ReduxObject,
    reducers,
    enhancers?
): Store<ReduxObject, AnyAction> {
    const combinedReducers = combineReducers<ReduxObject, AnyAction>(reducers);
    const rootReducer = (state: ReduxObject, action: Action): ReduxObject => {
        if ([USER_LOGIN, USER_LOGOUT, CONTEXT_SWITCH].some(it => it === action.type)) {
            state = initObject();
        }
        return combinedReducers(state, action);
    };
    return createStore<ReduxObject, AnyAction, {}, {}>(rootReducer, initialObject, composeWithDevTools(enhancers));
}

export const responsive = createResponsiveStateReducer(
    responsiveBP,
    {infinity: "xxl"}
);

export const store = configureStore(initObject(), {
    activity,
    dashboard,
    analyses,
    applications,
    header,
    status,
    sidebar,
    uploader,
    notifications,
    simpleSearch,
    detailedFileSearch,
    detailedApplicationSearch,
    fileInfo,
    hookStore,
    ...AppRedux.reducers,
    avatar,
    loading,
    tasks: TaskRedux.reducer,
    project: ProjectRedux.reducer,
    responsive: createResponsiveStateReducer(
        responsiveBP,
        {infinity: "xxl"}),
}, responsiveStoreEnhancer);

function loading(state = false, action: {type: string}): boolean {
    switch (action.type) {
        case "LOADING_START":
            return true;
        case "LOADING_END":
            return false;
        default:
            return state;
    }
}
