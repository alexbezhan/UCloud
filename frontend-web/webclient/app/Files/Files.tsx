import {RouteComponentProps} from "react-router";
import {useEffect} from "react";
import {getQueryParamOrElse} from "Utilities/URIUtilities";
import {Cloud} from "Authentication/SDUCloudObject";
import {Dispatch} from "redux";
import {setPrioritizedSearch, setRefreshFunction} from "Navigation/Redux/HeaderActions";
import {setActivePage, setLoading, updatePageTitle} from "Navigation/Redux/StatusActions";
import {SidebarPages} from "ui-components/Sidebar";
import {connect} from "react-redux";
import {FileTable} from "Files/FileTable";
import * as React from "react";
import {fileTablePage} from "Utilities/FileUtilities";
import {defaultVirtualFolders} from "Files/VirtualFileTable"
import {defaultFileOperations} from "Files/FileOperations";

interface FilesOperations {
    onInit: () => void
    refreshHook: (register: boolean, fn: () => void) => void,
    setLoading: (loading: boolean) => void
}

type FilesProps = RouteComponentProps & FilesOperations;

const Files: React.FunctionComponent<FilesProps> = props => {
    const urlPath = getQueryParamOrElse(props, "path", Cloud.homeFolder);
    useEffect(() => props.onInit(), []);

    return <FileTable
        {...defaultVirtualFolders()}
        fileOperations={defaultFileOperations.filter(it => it.text !== "View Parent")}
        embedded={false}
        onFileNavigation={path => props.history.push(fileTablePage(path))}
        path={urlPath}
        onLoadingState={props.setLoading}
        refreshHook={props.refreshHook}
    />;
};

const mapDispatchToProps = (dispatch: Dispatch): FilesOperations => ({
    onInit: () => {
        dispatch(setPrioritizedSearch("files"));
        dispatch(updatePageTitle("Files"));
        dispatch(setActivePage(SidebarPages.Files));
    },
    refreshHook: (register, fn) => {
        if (register) {
            dispatch(setRefreshFunction(fn));
        } else {
            dispatch(setRefreshFunction());
        }
    },
    setLoading: loading => dispatch(setLoading(loading))
});

export default connect(null, mapDispatchToProps)(Files);

