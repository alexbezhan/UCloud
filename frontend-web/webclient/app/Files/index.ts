import {ResponsiveReduxObject, SensitivityLevelMap} from "DefaultObjects";
import * as React from "react";
import {ClearRefresh, Operation, Page, PredicatedOperation} from "Types";
import {RouterLocationProps} from "Utilities/URIUtilities";
import {Times} from "./Redux/DetailedFileSearchActions";

export enum SortOrder {
    ASCENDING = "ASCENDING",
    DESCENDING = "DESCENDING"
}

export type FileType = "FILE" | "DIRECTORY" | "FAVFOLDER" | "SHARESFOLDER" | "TRASHFOLDER" | "RESULTFOLDER" |
    "SHARED_FS";

export interface File {
    fileType: FileType;
    path: string;
    creator: string | null;
    fileId: string | null;
    createdAt: number | null;
    modifiedAt: number | null;
    ownerName: string | null;
    size: number | null;
    acl: Acl[] | null;
    favorited: boolean | null;
    sensitivityLevel: SensitivityLevelMap | null;
    ownSensitivityLevel: SensitivityLevelMap | null;
    mockTag?: string;
}

export interface Acl {
    entity: string;
    rights: string[];
    group: boolean;
}

// FIXME: SortBy is subset of {FileResource}
export enum SortBy {
    FILE_TYPE = "fileType",
    PATH = "path",
    CREATED_AT = "createdAt",
    MODIFIED_AT = "modifiedAt",
    SIZE = "size",
    ACL = "acl",
    SENSITIVITY_LEVEL = "sensitivityLevel",
}

export enum FileResource {
    FAVORITED = "favorited",
    FILE_TYPE = "fileType",
    PATH = "path",
    CREATED_AT = "createdAt",
    MODIFIED_AT = "modifiedAt",
    OWNER_NAME = "ownerName",
    SIZE = "size",
    ACL = "acl",
    SENSITIVITY_LEVEL = "sensitivityLevel",
    OWN_SENSITIVITY_LEVEL = "ownSensitivityLevel",
    FILE_ID = "fileId",
    CREATOR = "creator"
}

export type FilesProps = FilesStateProps & FilesOperations & RouterLocationProps;

export interface FilesStateProps {
    path: string
    page: Page<File>
    loading: boolean
    fileSelectorShown: boolean
    fileSelectorLoading: boolean
    disallowedPaths: string[]
    fileSelectorCallback: (file: File) => void
    fileSelectorPath: string
    fileSelectorPage: Page<File>
    fileSelectorIsFavorites: boolean
    sortBy: SortBy
    sortOrder: SortOrder
    error?: string
    fileSelectorError?: string
    favFilesCount: number
    renamingCount: number
    sensitivityCount: number
    aclCount: number
    fileCount: number
    leftSortingColumn: SortBy
    rightSortingColumn: SortBy
    invalidPath: boolean
    responsive?: ResponsiveReduxObject
}

export interface FilesOperations extends ClearRefresh {
    onInit: () => void
    onFileSelectorErrorDismiss: () => void
    dismissError: () => void
    fetchFiles: (path: string, itemsPerPage: number, pageNumber: number, sortOrder: SortOrder, sortBy: SortBy, attrs: FileResource[], index?: number) => void
    fetchPageFromPath: (path: string, itemsPerPage: number, sortOrder: SortOrder, sortBy: SortBy, attrs: FileResource[]) => void;
    fetchSelectorFiles: (path: string, pageNumber: number, itemsPerPage: number) => void
    fetchFileSelectorFavorites: (pageNumber: number, itemsPerPage: number) => void
    setFileSelectorCallback: (callback: (file: File) => void) => void
    checkFile: (checked: boolean, path: string) => void
    setLoading: (loading: boolean) => void
    updateFiles: (files: Page<File>) => void
    updatePath: (path: string) => void
    showFileSelector: (open: boolean) => void
    checkAllFiles: (checked: boolean) => void
    setDisallowedPaths: (disallowedPaths: string[]) => void
    showUploader: () => void
    setUploaderCallback: (callback: (s: string) => void) => void
    createFolder: () => void
}

export interface FileSelectorProps {
    initialPath?: string
    onFileSelect: (file: {path: string} | null) => void
    canSelectFolders?: boolean
    onlyAllowFolders?: boolean
    trigger: React.ReactNode
    visible: boolean
    disallowedPaths?: string[]
}

export enum FileSource {
    HOME,
    FAVORITES,
    SHARES
}

export interface FilesTableProps {
    onNavigationClick?: (path: string) => void
    canNavigateFiles?: boolean
    sortOrder: SortOrder
    onDropdownSelect?: (sortOrder: SortOrder, sortBy: SortBy, index?: number) => void
    sortingColumns: SortBy[]
    files: File[]
    masterCheckbox?: React.ReactNode
    sortingIcon?: (name: SortBy) => "arrowUp" | "arrowDown" | undefined
    sortFiles: (sortOrder: SortOrder, sortBy: SortBy) => void
    onRenameFile?: (key: number, file: File, name: string) => void
    onCreateFolder?: (key: number, name: string) => void
    onCheckFile: (c: boolean, f: File) => void
    refetchFiles: () => void
    sortBy: SortBy
    onFavoriteFile?: (f: File[]) => void
    fileOperations: FileOperation[]
    responsive: ResponsiveReduxObject
    notStickyHeader?: boolean
}

export interface FilesTableHeaderProps {
    toSortingIcon?: (s: SortBy) => "arrowUp" | "arrowDown" | undefined
    sortFiles?: (sortOrder: SortOrder, sortBy: SortBy) => void
    sortOrder: SortOrder
    sortBy: SortBy
    masterCheckbox?: React.ReactNode
    sortingColumns: SortBy[]
    onDropdownSelect?: (sortOrder: SortOrder, sortBy: SortBy, index: number) => void
    customEntriesWidth?: string
    notStickyHeader?: boolean
}

export interface FilenameAndIconsProps {
    size?: number | string
    canNavigateFiles: boolean
    file: File
    hasCheckbox: boolean
    onRenameFile?: (key: number, file: File, name: string) => void
    onCheckFile?: (c: boolean) => void
    onFavoriteFile?: (files: File[]) => void
    onNavigationClick?: (path: string) => void
}

export interface MoveCopyOperations {
    showFileSelector: (show: boolean) => void
    setDisallowedPaths: (paths: string[]) => void
    setFileSelectorCallback: (callback: (file: File) => void) => void
    fetchPageFromPath: (path: string) => void
    fetchFilesPage: (path: string) => void
}

export interface FileOptionsProps {
    files: File[]
    fileOperations: FileOperation[]
}

export interface SortByDropdownProps {
    currentSelection: SortBy
    sortOrder: SortOrder
    onSelect: (sortorder: SortOrder, s: SortBy) => void
    asDropdown: boolean
    isSortedBy: boolean
}

export type FileOperation = Operation<File> | PredicatedOperation<File>

export interface ContextButtonsProps {
    createFolder: () => void;
    showUploader: () => void;
    inTrashFolder: boolean;
    toHome: () => void;
}


export interface DetailedFileSearchOperations {
    toggleHidden: () => void;
    addExtensions: (ext: string[]) => void;
    removeExtensions: (ext: string[]) => void;
    toggleFolderAllowed: () => void;
    toggleFilesAllowed: () => void;
    toggleIncludeShares: () => void;
    addSensitivity: (sensitivity: SensitivityLevel) => void;
    removeSensitivity: (sensitivity: SensitivityLevel[]) => void;
    addTags: (tags: string[]) => void;
    removeTags: (tags: string[]) => void;
    setFilename: (filename: string) => void;
    setLoading: (loading: boolean) => void;
    setTimes: (times: Times) => void;
}

export type DetailedFileSearchStateProps = DetailedFileSearchReduxState & DetailedFileSearchOperations;

export type SensitivityLevel = "Private" | "Confidential" | "Sensitive";

export interface DetailedFileSearchReduxState {
    hidden: boolean
    allowFolders: boolean
    allowFiles: boolean
    fileName: string
    extensions: Set<string>
    tags: Set<string>
    sensitivities: Set<SensitivityLevel>
    createdBefore?: Date
    createdAfter?: Date
    modifiedBefore?: Date
    modifiedAfter?: Date
    includeShares: boolean
    error?: string
    loading: boolean
}

export type ContextBarProps = ContextButtonsProps & FileOptionsProps & {invalidPath: boolean}

export type PossibleTime = "createdBefore" | "createdAfter" | "modifiedBefore" | "modifiedAfter";

export interface ResponsiveTableColumnProps extends SortByDropdownProps {
    iconName?: "arrowUp" | "arrowDown"
    minWidth?: number
    notSticky?: boolean
}

export type AdvancedSearchRequest = {
    fileName?: string
    extensions?: String[]
    fileTypes: [FileType?, FileType?]
    createdAt?: {after?: number, before?: number}
    modifiedAt?: {after?: number, before?: number}
    sensitivity?: SensitivityLevel[]
    includeShares?: Boolean
    itemsPerPage?: number
    page?: number
};
