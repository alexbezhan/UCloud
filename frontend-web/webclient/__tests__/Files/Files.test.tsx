import {createBrowserHistory} from "history";
import * as React from "react";
import {Provider} from "react-redux";
import {Router} from "react-router";
import {create} from "react-test-renderer";
import {ThemeProvider} from "styled-components";
import Files from "../../app/Files/Files";
import theme from "../../app/ui-components/theme";
import {store} from "../../app/Utilities/ReduxUtilities";

test("Mount Files component", () => {
    const history = createBrowserHistory();
    history.push("app/files?path=%2Fhome%2Fjonas%40hinchely.dk");
    expect(create(
        <Provider store={store}>
            <ThemeProvider theme={theme}>
                <Router history={history}>
                    <Files />
                </Router>
            </ThemeProvider>
        </Provider>
    )).toMatchSnapshot();
});
