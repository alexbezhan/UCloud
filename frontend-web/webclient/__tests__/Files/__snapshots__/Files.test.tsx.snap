// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mount Files component 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 160px;
}

.c27 {
  width: 32px;
  height: 32px;
  margin: 20px auto;
  -webkit-animation-name: jkjquP;
  animation-name: jkjquP;
}

.c27 > svg path:nth-child(1) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.04s;
  animation-delay: -0.04s;
}

.c27 > svg path:nth-child(2) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.08s;
  animation-delay: -0.08s;
}

.c27 > svg path:nth-child(3) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.12s;
  animation-delay: -0.12s;
}

.c27 > svg path:nth-child(4) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.16s;
  animation-delay: -0.16s;
}

.c27 > svg path:nth-child(5) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c27 > svg path:nth-child(6) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.24s;
  animation-delay: -0.24s;
}

.c27 > svg path:nth-child(7) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.28s;
  animation-delay: -0.28s;
}

.c27 > svg path:nth-child(8) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.32s;
  animation-delay: -0.32s;
}

.c27 > svg path:nth-child(9) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.36s;
  animation-delay: -0.36s;
}

.c27 > svg path:nth-child(10) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.4s;
  animation-delay: -0.4s;
}

.c27 > svg path:nth-child(11) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.44s;
  animation-delay: -0.44s;
}

.c27 > svg path:nth-child(12) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.48s;
  animation-delay: -0.48s;
}

.c27 > svg path:nth-child(13) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.52s;
  animation-delay: -0.52s;
}

.c27 > svg path:nth-child(14) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.56s;
  animation-delay: -0.56s;
}

.c27 > svg path:nth-child(15) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.6s;
  animation-delay: -0.6s;
}

.c27 > svg path:nth-child(16) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.64s;
  animation-delay: -0.64s;
}

.c27 > svg path:nth-child(17) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.68s;
  animation-delay: -0.68s;
}

.c27 > svg path:nth-child(18) {
  -webkit-animation: jkjquP 0.72s linear infinite;
  animation: jkjquP 0.72s linear infinite;
  -webkit-animation-delay: -0.72s;
  animation-delay: -0.72s;
}

.c0 {
  margin-left: 190px;
  padding-top: 48px;
  padding-bottom: 14px;
  padding-left: 14px;
  padding-right: 0;
  background-color: #fff;
}

.c8 {
  margin-left: auto;
}

.c16 {
  padding-left: 5px;
  padding-right: 5px;
  height: calc(100% - 20px);
}

.c22 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c23 {
  padding-top: 96px;
  padding-right: 240px;
}

.c19 {
  font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 5px;
  background-color: var(--blue,#f00);
  color: var(--white,#f00);
  border-width: 0;
  border-style: solid;
  -webkit-transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  font-size: 14px;
  padding: 9.5px 18px;
}

.c19:disabled {
  opacity: 0.25;
}

.c19:focus {
  outline: none;
}

.c19:hover {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c21 {
  font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 5px;
  background-color: var(--blue,#f00);
  color: var(--white,#f00);
  border-width: 0;
  border-style: solid;
  -webkit-transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  transition: cubic-bezier(0.5,0,0.25,1) 60ms;
  font-size: 14px;
  padding: 9.5px 18px;
  color: var(--blue,#f00);
  border: 2px solid var(--blue,#f00);
  border-radius: 5px;
  background-color: transparent;
  font-size: 14px;
  padding: 7.5px 18px;
}

.c21:disabled {
  opacity: 0.25;
}

.c21:focus {
  outline: none;
}

.c21:hover {
  -webkit-transform: scale(1.03);
  -ms-transform: scale(1.03);
  transform: scale(1.03);
}

.c21:hover {
  border: 2px solid;
  background-color: transparent;
  -webkit-transition: ease 0.1s;
  transition: ease 0.1s;
}

.c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: pointer;
}

.c13 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: inherit;
  margin-left: .7em;
}

.c26 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  vertical-align: middle;
  cursor: pointer;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  margin-right: .4em;
}

.c5 {
  cursor: pointer;
  font-size: 11px;
}

.c6 {
  cursor: inherit;
  font-size: 24px;
  margin-left: 6px;
  margin-right: 6px;
}

.c11 {
  cursor: pointer;
}

.c12 {
  cursor: inherit;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 98%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 4px;
}

.c17 .c18,
.c17 .c20 {
  width: 100%;
  margin-bottom: 8px;
}

.c10 {
  position: relative;
  display: inline-block;
}

.c10:hover > div {
  display: block;
}

.c1 {
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 204px;
  padding-right: 14px;
  width: 100%;
  height: 96px;
  background-color: #fff;
  position: absolute;
  top: 48px;
  left: 0;
  position: fixed;
  z-index: 50;
}

.c15 {
  padding-top: 144px;
  padding-left: 14px;
  padding-right: 14px;
  width: 240px;
  height: 100%;
  position: absolute;
  top: 0;
  right: 0;
  overflow-y: auto;
  position: fixed;
  z-index: 45;
}

.c4 {
  text-align: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% - 180px);
}

.c7 > span {
  width: 1;
  font-size: 25px;
  display: inline-block;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c7 > span + span:before {
  padding: 0 8px;
  vertical-align: top;
  color: var(--text,#f00);
  content: "/";
}

.c7 > span {
  color: var(--text,#f00);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 > span:hover {
  cursor: pointer;
  color: var(--blue,#f00);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 > span:last-child:hover {
  color: var(--text,#f00);
  cursor: default;
}

.c25 {
  background-color: #fff;
  position: -webkit-sticky;
  position: sticky;
  top: 144px;
  z-index: 50;
}

@media screen and (max-width:511px) {
  .c14 {
    display: none;
  }
}

@media screen and (min-width:512px) and (max-width:639px) {
  .c14 {
    display: none;
  }
}

@media screen and (min-width:640px) and (max-width:767px) {
  .c14 {
    display: none;
  }
}

@media screen and (min-width:768px) and (max-width:1023px) {
  .c24 {
    display: none;
  }
}

@media screen and (min-width:1024px) and (max-width:1279px) {
  .c24 {
    display: none;
  }
}

@media screen and (min-width:1280px) {
  .c24 {
    display: none;
  }
}

<div
  className="c0"
>
  <div
    className="sc-AxgMl c1"
    height={96}
    width={1}
  >
    <div
      className="c2"
    >
      <div
        className=""
      >
        <svg
          className="c3"
          cursor="pointer"
          fill="currentcolor"
          height="30px"
          onClick={[Function]}
          test-tag="to_home"
          viewBox="0 0 148.167 127.257"
          width="30px"
        >
          <path
            d="M0 63.5l10.583 10.583 63.5-52.916 63.5 52.916L148.167 63.5 74.083 0z"
          />
          <path
            d="M74.083 28.149l-50.27 41.618v57.49h34.316V92.312c0-4.162 2.06-11.64 15.954-11.628 13.894.012 15.958 7.406 15.958 11.71v34.863h34.313v-57.49zM124.354 38.92l-19.553-17.224V5.252l19.553.04z"
          />
        </svg>
        <div
          className="c4"
        >
          <div
            className="c5"
            cursor="pointer"
            fontSize="11px"
            onClick={[Function]}
          >
            Home
          </div>
        </div>
      </div>
      <div
        className="c6"
        cursor="inherit"
        fontSize="24px"
      >
        |
      </div>
      <div
        className="c7"
      >
        <span
          title="Home of jonas@hinchely.dk"
        >
          Home of jonas@hinchely.dk
        </span>
      </div>
      <div
        className="c8"
      />
      <div
        className="c9"
      >
        <div
          className="c10"
          data-tag="dropdown"
        >
          <span
            className="c11"
            cursor="pointer"
            onClick={[Function]}
          >
            <span
              className="c12"
              cursor="inherit"
            >
               
              Files per page 25
            </span>
            <svg
              className="c13"
              clipRule="evenodd"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height=".7em"
              ml=".7em"
              viewBox="0 0 24 20"
              width=".7em"
            >
              <path
                d="M11.998 19.999c-.598 0-1.028-.61-1.028-.61L.365 2.268C-.453 1.02.153 0 1.703 0h20.593c1.554 0 2.156 1.02 1.34 2.27l-10.61 17.121c0-.002-.43.608-1.028.608z"
                fillRule="nonzero"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c14"
  >
    <div
      className="sc-AxgMl c15"
      data-tag="sidebar"
      height="100%"
      width={240}
    >
      <div
        className="c16"
        height="calc(100% - 20px)"
      >
        <div
          className="c17"
        >
          <button
            className="c18 c19"
            color="blue"
            cursor="pointer"
            data-tag="Upload Files-action"
            onClick={[Function]}
          >
            <span>
              Upload Files
            </span>
          </button>
          <button
            className="c18 c20 c21"
            color="blue"
            cursor="pointer"
            data-tag="New Folder-action"
            onClick={[Function]}
          >
            <span>
              New Folder
            </span>
          </button>
          <div
            className="c22"
          />
          <button
            className="c18 c20 c21"
            color="blue"
            onClick={[Function]}
          >
            Use your files locally (BETA)
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c23"
  >
    <div
      className="c24"
    >
      <div
        className="c16"
        height="calc(100% - 20px)"
      >
        <div
          className="c17"
        >
          <button
            className="c18 c19"
            color="blue"
            cursor="pointer"
            data-tag="Upload Files-action"
            onClick={[Function]}
          >
            <span>
              Upload Files
            </span>
          </button>
          <button
            className="c18 c20 c21"
            color="blue"
            cursor="pointer"
            data-tag="New Folder-action"
            onClick={[Function]}
          >
            <span>
              New Folder
            </span>
          </button>
          <div
            className="c22"
          />
          <button
            className="c18 c20 c21"
            color="blue"
            onClick={[Function]}
          >
            Use your files locally (BETA)
          </button>
        </div>
      </div>
    </div>
    <div
      className="c25"
    >
      <div
        className="c2"
      >
        <div
          className="c8"
        />
        <div
          className="c10"
          data-tag="dropdown"
        >
          <span
            className="c11"
            cursor="pointer"
            onClick={[Function]}
          >
            <svg
              className="c26"
              clipRule="evenodd"
              cursor="pointer"
              fill="currentcolor"
              fillRule="evenodd"
              height=".7em"
              mr=".4em"
              rotation={180}
              viewBox="0 0 18 24"
              width=".7em"
            >
              <path
                d="M9 24l9-10h-6V0H6v14h-6l9 10z"
              />
            </svg>
            Sort by: 
            Filename
            <svg
              className="c13"
              clipRule="evenodd"
              cursor="inherit"
              fill="currentcolor"
              fillRule="evenodd"
              height=".7em"
              ml=".7em"
              viewBox="0 0 24 20"
              width=".7em"
            >
              <path
                d="M11.998 19.999c-.598 0-1.028-.61-1.028-.61L.365 2.268C-.453 1.02.153 0 1.703 0h20.593c1.554 0 2.156 1.02 1.34 2.27l-10.61 17.121c0-.002-.43.608-1.028.608z"
                fillRule="nonzero"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div
      className="c27"
      height={32}
      width={32}
    >
      <svg
        height={32}
        id="svg"
        version="1.1"
        viewBox="0 0 220 220"
        width={32}
        xmlns="http://www.w3.org/2000/svg"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      >
        <path
          d="M80.296,0.12l14.673,54.761l40.088,-40.088l-54.761,-14.673Z"
        />
        <path
          d="M149.73,69.554l-14.673,-54.761l-40.088,40.088l54.761,14.673Z"
        />
        <path
          d="M189.818,29.466l-54.761,-14.673l14.673,54.761l40.088,-40.088Z"
        />
        <path
          d="M204.491,84.228l-14.673,-54.762l-40.088,40.088l54.761,14.674Z"
        />
        <path
          d="M164.403,124.316l40.088,-40.088l-54.761,-14.674l14.673,54.762Z"
        />
        <path
          d="M219.165,138.989l-14.674,-54.761l-40.088,40.088l54.762,14.673Z"
        />
        <path
          d="M179.077,179.077l-14.674,-54.761l54.762,14.673l-40.088,40.088Z"
        />
        <path
          d="M124.316,164.403l54.761,14.674l-14.674,-54.761l-40.087,40.087Z"
        />
        <path
          d="M138.989,219.165l40.088,-40.088l-54.761,-14.674l14.673,54.762Z"
        />
        <path
          d="M84.228,204.491l54.761,14.674l-14.673,-54.762l-40.088,40.088Z"
        />
        <path
          d="M69.554,149.73l14.674,54.761l40.088,-40.088l-54.762,-14.673Z"
        />
        <path
          d="M29.466,189.818l54.762,14.673l-14.674,-54.761l-40.088,40.088Z"
        />
        <path
          d="M14.793,135.057l14.673,54.761l40.088,-40.088l-54.761,-14.673Z"
        />
        <path
          d="M54.881,94.969l-40.088,40.088l54.761,14.673l-14.673,-54.761Z"
        />
        <path
          d="M0.12,80.296l14.673,54.761l40.088,-40.088l-54.761,-14.673Z"
        />
        <path
          d="M40.208,40.208l-40.088,40.088l54.761,14.673l-14.673,-54.761Z"
        />
        <path
          d="M94.969,54.881l-54.761,-14.673l14.673,54.761l40.088,-40.088Z"
        />
        <path
          d="M80.296,0.12l-40.088,40.088l54.761,14.673l-14.673,-54.761Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
